sysutils/scp-backup/src/opnsense/mvc/app/controllers/OPNsense/ScpBackup/Api/GeneralController.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+/**
+ *    Copyright (C) 2018 David Harrigan
+ *    Copyright (C) 2015 - 2017 Deciso B.V.
+ *
+ *    All rights reserved.
+ *
+ *    Redistribution and use in source and binary forms, with or without
+ *    modification, are permitted provided that the following conditions are met:
+ *
+ *    1. Redistributions of source code must retain the above copyright notice,
+ *       this list of conditions and the following disclaimer.
+ *
+ *    2. Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *
+ *    THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ *    INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+ *    AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *    AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ *    OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *    POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+namespace OPNsense\ScpBackup\Api;
+
+use \OPNsense\Base\ApiMutableModelControllerBase;
+use \OPNsense\Core\Backend;
+use \OPNsense\Core\Config;
+use \OPNsense\Cron\Cron;
+use \OPNsense\ScpBackup\General;
+
+class GeneralController extends ApiMutableModelControllerBase
+{
+    static protected $internalModelName = 'general';
+    static protected $internalModelClass = 'OPNsense\ScpBackup\General';
+
+    public function getAction()
+    {
+        $result = array();
+        if ($this->request->isGet()) {
+            $mdlGeneral = $this->getModel();
+            $publicKey = fopen("/conf/sshd/ssh_host_rsa_key.pub", "r");
```

Comment:
this might be the wrong key - it is only or the server to show its identity to the client. It should not be used for that.

Reply:
I believe it is the correct key, as this is the only available to me - "root" on OPNsense doesn't have a default generated public/private keypair - Fitch pointed me to this particular file.